include/zephyr/dt-bindings/pinctrl/numaker_m46x_pinctrl.h

```diff
@@ -0,0 +1,2154 @@
+/*
+ * Copyright (c) 2023 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_PINCTRL_NUMAKER_M46X_PINCTRL_H
+#define ZEPHYR_INCLUDE_DT_BINDINGS_PINCTRL_NUMAKER_M46X_PINCTRL_H
+
+/* Beginning of M460 BSP sys_reg.h pin-mux module copy */
+
+#define SYS_GPA_MFP0_PA0MFP_Pos          (0)
+#define SYS_GPA_MFP0_PA1MFP_Pos          (8)
+#define SYS_GPA_MFP0_PA2MFP_Pos          (16)
+#define SYS_GPA_MFP0_PA3MFP_Pos          (24)
```

Comment:
these need to be namespaced as other pinctrl dt-bindings, e.g. `NUMAKER_M46X_...`

Reply:
@gmarull ,
Except xxx_pinctrl.h, xxx_clock.h seems also be requested.
After checking, if add namespace "NUMAKER" for each macro in numaker_m46x_clock.h, the macro will be much longer & not easy read, like as: 
```
#define NUMAKER_UART0_MODULE                                                                               \
	(NUMAKER_MODULE_APBCLK_ENC(1UL) | NUMAKER_MODULE_IP_EN_Pos_ENC(NUMAKER_CLK_APBCLK0_UART0CKEN_Pos) |                \
	 NUMAKER_MODULE_CLKSEL_ENC(1UL) | NUMAKER_MODULE_CLKSEL_Msk_ENC(3UL) | NUMAKER_MODULE_CLKSEL_Pos_ENC(24UL) |       \
	 NUMAKER_MODULE_CLKDIV_ENC(0UL) | NUMAKER_MODULE_CLKDIV_Msk_ENC(0x0FUL) | NUMAKER_MODULE_CLKDIV_Pos_ENC(8UL))
```
Could you accept the above modification ?

Reply:
According to @gmarull instructions, modified `numaker_m46x_clock.h/numaker_m46x_reset.h/numaker-m46x-pinctrl.h` to be namespaced as NUMAKER_....

Also rebase to resolve merge conflcit of `pinctrl/Kconfig & CMakeLists.txt`.
Please take a look.

Reply:
@gmarull , @dcpleung ,

Finished the namespace of dt-bindings, please take another look.

Reply:
@gmarull, 
not get your feedback over 3 weeks, 
is there any concern ?